src/main/java/org/opensearch/plugin/insights/core/exporter/LocalIndexExporter.java

```diff
@@ -180,12 +185,64 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Check mapping compatibility and perform bulk export with appropriate format
+     */
+    void checkMappingAndBulk(String indexName, List<SearchQueryRecord> records) {
```

Comment:
Right now we are checking the index mapping before every bulk, but most times this is unnecessary work. I wonder if it would make more sense to first try the bulk request with useObjectSource = true. Only if the bulk fails with `MapperException` then we optionally do the "check mapping" flow and retry with useObjectSource = false.